### PR TITLE
 Added  fabric, secure channel and cluster specfic traces for better tracing.

### PR DESCRIPTION
**Pull request by shripad621git**
Change Overview:
1.Added some more traces to  CASE and PASE session files.
2.Added fabric related traces.
3.Added cluster specific traces.

### CODE REVIEW

File: src/app/clusters/color-control-server/color-control-server.cpp [hunk 1]
```diff
@@ -1798,6 +1804,7 @@ bool ColorControlServer::moveSaturationCommand(app::CommandHandler * commandObj,
 bool ColorControlServer::moveToSaturationCommand(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
                                                  const Commands::MoveToSaturation::DecodableType & commandData)
 {
+    MATTER_TRACE_SCOPE("moveToSaturatio", "ColorControl");
```
Comment (bzbarsky-apple) [inline src/app/clusters/color-control-server/color-control-server.cpp:1807]:
```suggestion
    MATTER_TRACE_SCOPE("moveToSaturation", "ColorControl");
```

File: src/app/clusters/operational-credentials-server/operational-credentials-server.cpp [hunk 2]
```diff
@@ -354,6 +354,7 @@ class OpCredsFabricTableDelegate : public chip::FabricTable::Delegate
     void OnFabricRemoved(const FabricTable & fabricTable, FabricIndex fabricIndex) override
     {
         ChipLogProgress(Zcl, "OpCreds: Fabric index 0x%x was removed", static_cast<unsigned>(fabricIndex));
+        MATTER_TRACE_INSTANT("FabricRemoved", "OperationalCredentials");
```
Comment (bzbarsky-apple) [inline src/app/clusters/operational-credentials-server/operational-credentials-server.cpp:357]: Should this tracing really be here, or in the fabric table?

File: src/app/clusters/operational-state-server/operational-state-server.cpp [hunk 3]
```diff
@@ -26,6 +26,7 @@
 #include <app/InteractionModelEngine.h>
 #include <app/reporting/reporting.h>
 #include <app/util/attribute-storage.h>
+#include <tracing/macros.h>
```
Comment (bzbarsky-apple) [inline src/app/clusters/operational-state-server/operational-state-server.cpp:29]: Where is this used in this file?

File: src/protocols/secure_channel/CASESession.cpp [hunk 4]
```diff
@@ -462,6 +464,8 @@ CASESession::PrepareForSessionEstablishment(SessionManager & sessionManager, Fab
                                             SessionEstablishmentDelegate * delegate, const ScopedNodeId & previouslyEstablishedPeer,
                                             Optional<ReliableMessageProtocolConfig> mrpLocalConfig)
 {
+
```
Comment (bzbarsky-apple) [inline src/protocols/secure_channel/CASESession.cpp:467]:
```suggestion
```

File: src/credentials/FabricTable.cpp [hunk 5]
```diff
@@ -605,19 +613,22 @@ CHIP_ERROR FabricTable::FetchICACert(FabricIndex fabricIndex, MutableByteSpan &
...
 {
+    MATTER_TRACE_SCOPE("FetchRootPubkey", "Fabric");
     const FabricInfo * fabricInfo = FindFabricWithIndex(fabricIndex);
     ReturnErrorCodeIf(fabricInfo == nullptr, CHIP_ERROR_INVALID_FABRIC_INDEX);
     return fabricInfo->FetchRootPubkey(outPublicKey);
 }
 
 CHIP_ERROR FabricTable::FetchCATs(const FabricIndex fabricIndex, CATValues & cats) const
 {
+
```
Comment (shubhamdp) [inline src/credentials/FabricTable.cpp:631]:
```suggestion
```

File: src/tracing/esp32_trace/esp32_tracing.cpp [hunk 6]
```diff
@@ -29,7 +29,7 @@ namespace Tracing {
 namespace Insights {
 namespace {
 
-constexpr size_t kPermitListMaxSize = 10;
+constexpr size_t kPermitListMaxSize = 20;
```
Comment (shubhamdp) [inline src/tracing/esp32_trace/esp32_tracing.cpp:32]: Can this be a configuration option? This can be a follow up.

**Review by shubhamdp — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by dhrishi — APPROVED** [review]

**Pull request merged**